packages/features/bookings/lib/handleNewBooking/getBookingData.ts

```diff
@@ -68,7 +68,12 @@ export async function getBookingData<T extends z.ZodType>({
     name: responses.name,
     email: responses.email,
     guests: responses.guests ? responses.guests : [],
-    location: responses.location?.optionValue || responses.location?.value || "",
+    location:
+      typeof responses.location === "object"
+        ? responses.location?.optionValue || responses.location?.value || ""
+        : typeof responses.location === "string"
+        ? responses.location
+        : "",
```

Comment:
```suggestion
    location: responses.location
      ? typeof responses.location === "object"
        ? responses.location.optionValue || responses.location.value
        : responses.location
      : "",
```

NIT: Since location can only be object or string, if not object then we shouldn't have to check if its a string.